community/kernel/src/main/java/org/neo4j/kernel/configuration/Config.java

```diff
@@ -82,6 +85,7 @@
     private final List<ConfigOptions> configOptions;
 
     private final Map<String,String> params = new CopyOnWriteHashMap<>(); // Read heavy workload
+    private final Set<String> secrets = new HashSet<>();
```

Comment:
You don't need to populate this, you can use `settingsMap` to fetch the setting.

Reply:
I wanted the obsfucateIfSecret method to have lowest cost, and thought a pre-populated hashmap of very small size (zero or one entry with current common cases) was the best. Otherwise we check a large collection every time.